src/NuGet.Indexing/ResponseFormatter.cs

```diff
@@ -346,7 +346,19 @@ public static void WriteStatsResult(JsonWriter jsonWriter, NuGetIndexSearcher se
             jsonWriter.WriteStartObject();
             WriteProperty(jsonWriter, "numDocs", searcher.IndexReader.NumDocs());
```

Comment:
I accept that this is outside of the scope of the PR, but wouldn't C# handle any complexities in type conversions through `WriteProperty<T>`'s use of generics? I don't think there would be any more code necessary than what I put here.

Let me fix the second code example though...
```
foreach (var diagnosticName in searcher.Diagnostics.Keys) {
    WriteProperty(jsonWriter, diagnosticName, searcher.Diagnostics[diagnosticName]);
}
```